classes/01class/exercises/c01-aws01/frdvo/README.md

```diff
@@ -0,0 +1,318 @@
+                    "MacAddress": "06:69:45:23:75:e8",
+                    "NetworkInterfaceId": "eni-0324037b1c35182ac",
+                    "OwnerId": "165765640813",
+                    "PrivateDnsName": "ip-172-31-36-240.ap-southeast-2.compute.internal",
+                    "PrivateIpAddress": "172.31.36.240",
+                    "PrivateIpAddresses": [
+                        {
+                            "Primary": true,
+                            "PrivateDnsName": "ip-172-31-36-240.ap-southeast-2.compute.internal",
+                            "PrivateIpAddress": "172.31.36.240"
+                        }
+                    ],
+                    "SourceDestCheck": true,
+                    "Status": "in-use",
+                    "SubnetId": "subnet-baaffaf3",
+                    "VpcId": "vpc-9d0f66fa",
+                    "InterfaceType": "interface"
+                }
+            ],
+            "RootDeviceName": "/dev/xvda",
+            "RootDeviceType": "ebs",
+            "SecurityGroups": [
+                {
+                    "GroupName": "SSHOpen",
+                    "GroupId": "sg-063fc9a232e4e2a60"
+                }
+            ],
+            "SourceDestCheck": true,
+            "StateReason": {
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "165765640813",
+    "ReservationId": "r-069bd5f33d2bc601a"
+}
+
+- Commands to connect to the first EC2 instance:
+**GET IP Address**
+`aws ec2 describe-instances --query "Reservations[].Instances[].[InstanceId, PublicIpAddress]"`
+[
+    [
+        "i-0cede1a195229c66b",
+        null
+    ],
+    [
+        "i-02f268a05455dc892",
+        "3.25.62.164"
+    ]
+]
+**CONNECT**
+ssh -i da.pem ec2-user@3.25.62.164
+ec2-user@3.25.62.164: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
+cat dajp.pem
+cp dajp.pem dajp.pem.old
+chmod u+w da.pem
+vim dajp.pem - trim the key only
+ssh -i da.pem ec2-user@3.25.62.164
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-36-240 ~]$ exit
+
+
+- Commands to create the second EC2 instance and any additional resource required:
+
+**Generate SSH Key for Internal Servers**
+`aws ec2 create-key-pair --key-name daint --output text > daint.pem`
```

Comment:
you can use a key to connect to multiple instances. there is no need to have a 1:1 relationship between keys and EC2 instances.
In any case, its not a problem and it should be fine. just bring you additional effort to manage the keys.

Reply:
I agree. I just watched the reference video. The instructor recommends one key for the jump box and another key for the internal instances (As you say, for the internal we can share the same key). If the jump box key gets compromised, the attacker can't access the internal instances.